services/thermal_mgr/thermal_mgr.c

```diff
@@ -43,18 +44,52 @@ void initThermalSystemManager(lm75bd_config_t *config) {
 
 error_code_t thermalMgrSendEvent(thermal_mgr_event_t *event) {
   /* Send an event to the thermal manager queue */
+  if (event == NULL) {
+    return ERR_CODE_INVALID_ARG;
+  }
+
+  if (thermalMgrQueueHandle == NULL) {
+    return ERR_CODE_INVALID_STATE;
+  }
+
+   if (xQueueSend(thermalMgrQueueHandle, event, 0) == errQUEUE_FULL) {
+    return ERR_CODE_QUEUE_FULL;
+  }
+
 
   return ERR_CODE_SUCCESS;
 }
 
 void osHandlerLM75BD(void) {
   /* Implement this function */
+  thermal_mgr_event_t event = {.type = THERMAL_MGR_EVENT_OS_INT};
+  thermalMgrSendEvent(&event);
 }
 
 static void thermalMgr(void *pvParameters) {
-  /* Implement this task */
+  thermal_mgr_event_t event;
+
   while (1) {
-    
+    if (xQueueReceive(thermalMgrQueueHandle, &event, portMAX_DELAY) == pdPASS) {
```

Comment:
Set the blocking time to be lower than portMAX_DELAY. The reason is we want thermalMgr to obtain from the queue if there is an event, not wait a long time for an event to pop up